tests/test_helper.go

```diff
@@ -0,0 +1,323 @@
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/kubernetes"
+)
+
+// common constants
+const (
+	E2ETestObjectName  = "e2e-object"
+	MigrationNamespace = "openshift-migration"
+	TestSecretType     = "Opaque"
+)
+
+// cluster constants
+const (
+	TestDestinationCluster = "host"
+	TestClusterSecret      = "e2eclustersecret"
+)
+
+// storage constants
+const (
+	TestStorageSecret = "e2estoragesecret"
+	ConfigNamespace   = "openshift-config"
+)
+
+// migrationcontroller constants
+const (
+	MigrationController = "migration-controller"
+)
+
+// environment variables
+const (
+	EXPOSEDREGISTRYPATH       = "EXPOSEDREGISTRYPATH"
+	SOURCEURL                 = "SOURCEURL"
+	SOURCECONFIG              = "SOURCECONFIG"
+	HOSTCONFIG                = "KUBECONFIG"
+	VELERO_PLUGIN_FQIN        = "VELERO_PLUGIN_FQIN"
+	MIG_CONTROLLER_IMAGE_FQIN = "MIG_CONTROLLER_IMAGE_FQIN"
+	RSYNC_TRANSFER_IMAGE_FQIN = "RSYNC_TRANSFER_IMAGE_FQIN"
+	MIG_NAMESPACE_LIMIT       = "MIG_NAMESPACE_LIMIT"
+	MIG_POD_LIMIT             = "MIG_POD_LIMIT"
+	CLUSTER_NAME              = "CLUSTER_NAME"
+	RESTIC_TIMEOUT            = "RESTIC_TIMEOUT"
+	MIGRATION_VELERO          = "MIGRATION_VELERO"
+	MIG_PV_LIMIT              = "MIG_PV_LIMIT"
+	VERSION                   = "VERSION"
+	AWSBUCKETNAME             = "AWSBUCKETNAME"
+	AWSSECRETKEY              = "AWSSECRETKEY"
+	AWSACCESSKEY              = "AWSACCESSKEY"
+	BACKUPSTORAGEPROVIDER     = "BACKUPSTORAGEPROVIDER"
+)
+
+func NewMigMigration(name string, planRef string, quiesce bool, stage bool) *migapi.MigMigration {
+	return &migapi.MigMigration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: MigrationNamespace,
+		},
+		Spec: migapi.MigMigrationSpec{
+			MigPlanRef: &v1.ObjectReference{
+				Namespace: MigrationNamespace,
+				Name:      planRef,
+			},
+			Stage:       stage,
+			QuiescePods: quiesce,
+		},
+	}
+}
+
+func NewMigPlan(namespaces []string, name string) *migapi.MigPlan {
+	return &migapi.MigPlan{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: MigrationNamespace,
+		},
+		Spec: migapi.MigPlanSpec{
+			Namespaces: namespaces,
+			SrcMigClusterRef: &v1.ObjectReference{
+				Namespace: MigrationNamespace,
+				Name:      E2ETestObjectName,
+			},
+			DestMigClusterRef: &v1.ObjectReference{
+				Namespace: MigrationNamespace,
+				Name:      TestDestinationCluster,
+			},
+			MigStorageRef: &v1.ObjectReference{
+				Name:      E2ETestObjectName,
+				Namespace: MigrationNamespace,
+			},
+		},
+	}
+}
+
+func NewMigStorage() (*migapi.MigStorage, *v1.Secret) {
+	return &migapi.MigStorage{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      E2ETestObjectName,
+				Namespace: MigrationNamespace,
+			},
+			Spec: migapi.MigStorageSpec{
+				BackupStorageConfig: migapi.BackupStorageConfig{
+					CredsSecretRef: &v1.ObjectReference{
+						Namespace: ConfigNamespace,
+						Name:      TestStorageSecret,
+					},
+					AwsBucketName: os.Getenv(AWSBUCKETNAME),
```

Comment:
I needed a zone to make the migstorage to ready, can we add `AWSREGION` and configure it here for completeness? 

NIT: if your bucket is provisioned in global mode, zone is not required, but in my case it was not.

Reply:
Ahh, Okay! You got it! Adding the variable